src/web/rustdoc.rs

```diff
@@ -549,9 +552,15 @@ mod test {
         {
             let link = elem.attributes.borrow().get("href").unwrap().to_string();
             assert_success(&link, web)?;
-            return Ok(link);
+            Ok(Some(link))
+        } else {
+            Ok(None)
         }
-        panic!("no redirect found for {}", path);
+    }
+
+    fn latest_version_redirect(path: &str, web: &TestFrontend) -> Result<String, failure::Error> {
+        try_latest_version_redirect(path, web)
+            .and_then(|v| v.ok_or_else(|| failure::format_err!("no redirect found for {}", path)))
```

Comment:
Instead of making `try_latest` a separate function, could you put this error in the `Ok(None)` branch above?

Reply:
This was from when I had a test that asserted that there would be no redirect, so separating the lack of redirect from an error finding a redirect was useful.